java/jshell.support/src/org/netbeans/modules/jshell/support/ShellSession.java

```diff
@@ -944,8 +945,34 @@ public void notifyClosed(JShellEnvironment env, boolean remote) {
     }
     
     public void reportErrorMessage(Throwable t) {
-        LOG.log(Level.INFO, "Error in JSHell", t);
-        reportShellMessage(buildErrorMessage(t));
+        StringBuilder processOutput = null;
+        VMStartException startException = findVMStartException(t);
+        if(startException != null) {
+            processOutput = new StringBuilder();
+            // VMStartException is raised when target VM fails to start. At time
+            // of writing com.sun.tools.jdi.AbstractLauncher does not read the
+            // process output, so diagnostic output is lost, try to recover it.
+            assert !startException.process().isAlive() : "Process expected to be not alive";
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            try {
+                startException.process().getInputStream().transferTo(baos);
+            } catch (IOException ex) {}
+            processOutput.append("Standard output:\n");
```

Comment:
L10n as it goes to UI ?

Reply:
Valid point - updated via 25ee1a49b5838031bd0d52294d66628ca43512a8